apps/lib/cmp_mock_srv.c

```diff
@@ -490,6 +490,43 @@ static OSSL_CMP_ITAV *process_genm_itav(mock_srv_ctx *ctx, int req_nid,
                 rsp = OSSL_CMP_ITAV_new_crls(res == 0 ? NULL : ctx->crlOut);
         }
         break;
+    case NID_id_it_certReqTemplate:
+        {
+            OSSL_CRMF_CERTTEMPLATE *reqtemp;
+            OSSL_CMP_ATAVS *keyspec = NULL;
+            X509_ALGOR *keyalg = NULL;
+
+            if ((reqtemp = OSSL_CRMF_CERTTEMPLATE_new()) == NULL)
+                return NULL;
+
+            if (!OSSL_CRMF_CERTTEMPLATE_fill(reqtemp, NULL, NULL,
+                                             X509_get_issuer_name(ctx->refCert),
+                                             NULL))
+                goto crt_err;
+
+            if ((keyspec = OSSL_CMP_ATAVS_new()) == NULL)
+                goto crt_err;
+
+            if ((keyalg = X509_ALGOR_new()) == NULL)
+                goto crt_err;
+
+            (void)X509_ALGOR_set0(keyalg, OBJ_nid2obj(NID_X9_62_id_ecPublicKey),
+                                  V_ASN1_UNDEF, NULL); /* cannot fail */
+
+            if (!sk_OSSL_CMP_ATAV_push(keyspec, OSSL_CMP_ATAV_new_algId(keyalg))
+                || !sk_OSSL_CMP_ATAV_push(keyspec,
+                                          OSSL_CMP_ATAV_new_rsaKeyLen(4096)))
+                goto crt_err;
```

Comment:
If I use `OSSL_CMP_ATAV_push1()` then 2 new variable for `OSSL_CMP_ATAV_new_algId` and `OSSL_CMP_ATAV_new_rsaKeyLen` need to be handled.

Reply:
Two local variables unfortunately are needed anyway, because 
non-NULL result of `OSSL_CMP_ATAV_new_algId(keyalg)` and `OSSL_CMP_ATAV_new_rsaKeyLen(4096)` has to be checked, which the code so far does not.
```
alg = OSSL_CMP_ATAV_new_algId(keyalg);
len = OSSL_CMP_ATAV_new_rsaKeyLen(4096);
ok = OSSL_CMP_ATAV_push1(&keyspec, alg)
     && OSSL_CMP_ATAV_push1(&keyspec,len));
```
does this check internally.
Yet what is needed when using `OSSL_CMP_ATAV_push1()` is to free both `alg` and `len`,
regardless if `ok` holds or not.